### PR TITLE
Update to Otel 1.6

### PR DESCRIPTION
**Pull request by anuraaga**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: awsagentprovider/build.gradle.kts [hunk 1]
```diff
@@ -24,7 +24,7 @@ base {
 
 dependencies {
   compileOnly("io.opentelemetry.javaagent:opentelemetry-javaagent-extension-api")
-  compileOnly("io.opentelemetry:opentelemetry-sdk-extension-autoconfigure")
+  compileOnly("io.opentelemetry:opentelemetry-sdk-extension-autoconfigure-spi")
```
Comment (willarmiros) [inline awsagentprovider/build.gradle.kts:27]: Is this a breaking change from upstream?

**Review by willarmiros — APPROVED** [review]
LGTM

File: awsagentprovider/build.gradle.kts (hunk 1 above)
Comment (anuraaga) [reply awsagentprovider/build.gradle.kts:27]: The SPI interfaces were moved to a separate artifact, which would also be available through the `autoconfigure` package. But we don't need anything but the interfaces so pointed at the more precise one here.

Comment (willarmiros) [reply awsagentprovider/build.gradle.kts:27]: Ah I see it migrated from alpha. Makes sense thanks

**Pull request merged**